test/integration/filters/eds_ready_filter.cc

```diff
@@ -17,10 +17,14 @@ namespace Envoy {
 // responds OK to all requests if it is.
 class EdsReadyFilter : public Http::PassThroughFilter {
 public:
+  // TODO(jmarantz): this const-cast of the symbol-table is needed only because
```

Comment:
This seems like an anti-pattern. What's the proper fix here?

Reply:
Once we add the const method to get a StatName from a const SymbolTable that @jmarantz mentions in the comment, this problem goes away.

Reply:
The easiest fix is to make Api::rootScope() return a non-const scope. However I think the spirit of your concern expressed in the comments in https://github.com/envoyproxy/envoy/issues/6706 is justified.

I was thinking that the right fix is to define something like:

   StoragePtr SymbolTable::encodeIfExists(absl::string_view name);

which would return nullptr if name has never been encoded before, otherwise would return some storage that now needs to be managed, e.g.  by constructing a StatNameManagedStorage, out of which a StatName can be accessed.

But this might be a little cumbersome to use, because "cluster.cluster_0.membership_healthy" may be created after the filter is constructed, which is why findGauge() is run during request-processing.

So now I'm thinking of just removing the TODO -- the change here is only the name-change.

If we cared about how fast this filter was, though, we should at least remember the Gauge object and not search for it by name on every request. But since this is a in the test/... hierarchy we probably don't care.

Reply:
I see, fare enough.

Reply:
After reflecting on this I found a better solution and will do a PR once this one passes CI and merges.

 * Get the non-const symbol-table from `filter_factory.scopee().symbolTable()`
 * pass the `SymbolTable&` separate from the root-scope as a second constructor arg to `EdsReadyFilter`